### ISSUE TITLE
Clients may enclose cookie values in double quotes

### ISSUE THREAD

**Issue by albe** [neos/flow-development-collection#2133]
According to RFC 6265 https://tools.ietf.org/html/rfc6265#section-4.1.1 a cookie value may be enclosed in double quotes.

Apparently, Android clients for example make use of this:
https://stackoverflow.com/questions/19921514/android-basicclientcookie-cookiestore-adds-quotes-around-cookie-value

Currently, our `Cookie` class implementation does not strip double quotes from cookie values, therefore this may lead to invalid values being received, like this:
```
InvalidArgumentException: ""YsWSZUIQZwH4wO4Ei2BIhJzayT8PxEZl"" is not a valid cache entry identifier.
#11 /application/Packages/Libraries/neos/cache/Classes/Frontend/VariableFrontend.php(87): get
#10 /application/Packages/Framework/Neos.Flow/Classes/ObjectManagement/DependencyInjection/DependencyProxy.php(99): __call
#9 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Session_SessionManager.php(70): initializeCurrentSessionFromCookie
#8 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Session_Http_SessionRequestComponent.php(62): handle
#7 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Component_ComponentChain.php(62): handle
#6 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Component_ComponentChain.php(62): handle
#5 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Middleware_ComponentChainMiddleware.php(54): process
#4 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Middleware_MiddlewaresChain.php(78): process
#3 /application/Data/Temporary/Production/SubContextBeach/SubContextInstance/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Middleware_MiddlewaresChain.php(92): handle
#2 /application/Packages/Framework/Neos.Flow/Classes/Http/RequestHandler.php(102): handleRequest
#1 /application/Packages/Framework/Neos.Flow/Classes/Core/Bootstrap.php(112): run
#0 index.php(37): null
```

We need to trim double quotes from the cookie value in order to behave correctly for such clients.
https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Http/Cookie.php#L183

**Comment by sorenmalling** [comment]
Would 
`trim($str, '"');`
be sufficient for his?

**Comment by albe** [comment]
Would remove multiple quotes, not just a single pair.
e.g.
```
"double quote start
double quote end"
"double quote both"
""double double quote""
```
https://regex101.com/r/z6GeTs/2
(thx @bwaidelich)

`$cookieValue = preg_replace('/^"(.*)"$/', '$1', $cookieValue);` would work

**Comment by DavidSporer** [comment]
I just tested this change at the place that @albe suggested but it seems that this change is not enough.
Im not sure, however, where to fix this best.
It works if I remove the double quotes in the constructor as well:
https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Http/Cookie.php#L150
Like this:
`$this->value = $value === null ? null : preg_replace('/^"(.*)"$/', '$1', $value);`

Is this the right place to do this?

**Comment by DavidSporer** [comment]
Ah ok, I created the PR before your edit.
Sounds legit to keep the value but I think this case needs to be addressed. Let me know if I can help.
I think we'll add the commit as a patch for now to solve the issue for our users.

**Comment by bwaidelich** [comment]
Fixed with #2138
